src/hierarchies/mfa_hierarchy.h

```diff
@@ -0,0 +1,132 @@
+#ifndef BAYESMIX_HIERARCHIES_MFA_HIERARCHY_H_
+#define BAYESMIX_HIERARCHIES_MFA_HIERARCHY_H_
+
+#include <google/protobuf/stubs/casts.h>
+
+#include <Eigen/Dense>
+#include <memory>
+#include <vector>
+
+#include "algorithm_state.pb.h"
+#include "base_hierarchy.h"
+#include "hierarchy_id.pb.h"
+#include "hierarchy_prior.pb.h"
+
+//! Mixture of Factor Analysers hierarchy for multivariate data.
+
+//! This class represents a hierarchical model where data are distributed
+
+namespace MFA {
+//! Custom container for State values
+struct State {
+  Eigen::VectorXd mu, psi;
+  Eigen::MatrixXd eta, lambda;
+  Eigen::MatrixXd psi_inverse;
```

Comment:
I think using a full matrix for `psi_inverse` is kind of a waste. There is a `DiagonalMatrix` class for what you want in Eigen: http://eigen.tuxfamily.org/dox/classEigen_1_1DiagonalMatrix.html